js/src/utils/external.ts

```diff
@@ -30,49 +30,46 @@ export function sendProcessReq(info: {
   }
 
   try {
-    // Use node-fetch for making HTTP requests
     const url = new URL(info.url);
-    // eslint-disable-next-line @typescript-eslint/no-require-imports
-    const { spawn } = require("child_process");
-    const child = spawn("node", [
+    const protocol = url.protocol === "https:" ? "https" : "http";
+    const port = url.port || (url.protocol === "https:" ? 443 : 80);
+
+    const args = [
       "-e",
       `
-        const http = require('${url.protocol === "https:" ? "https" : "http"}');
-        const options = {
-          hostname: '${url.hostname}',
-          path: '${url.pathname}${url.search}',
-          port: ${url.port || (url.protocol === "https:" ? 443 : 80)},
-          method: '${info.method}',
-          headers: ${JSON.stringify(info.headers)}
-        };
+      const http = require('${protocol}');
+      const options = {
+        hostname: '${url.hostname}',
```

Comment:
@himanshu-dixit it doesn't need template strings. these can be used as variable directly

Reply:
don't think so! we're spawning this as new process in case of error.